apps/client/src/features/app-settings/panel/integrations-panel/HttpIntegrations.tsx

```diff
@@ -32,51 +97,76 @@ export default function HttpIntegrations() {
-                    <Input size='sm' variant='ontime-filled' value={integration.message} />
-                  </td>
-                  <td>
-                    <IconButton
-                      size='sm'
-                      variant='ontime-ghosted'
-                      color='#FA5656' // $red-500
-                      icon={<IoTrash />}
-                      aria-label='Delete entry'
-                    />
-                  </td>
+          {fields.length > 0 && (
+            <Panel.Table>
+              <thead>
+                <tr>
+                  <th>Enabled</th>
+                  <th>Cycle</th>
+                  <th className={style.fullWidth}>Message</th>
+                  <th />
                 </tr>
-              ))}
-            </tbody>
-          </Panel.Table>
+              </thead>
+              <tbody>
+                {fields.map((integration, index) => {
+                  // @ts-expect-error -- not sure why it is not finding the type, it is ok
+                  const maybeError = errors.subscriptions?.[index]?.message?.message;
+                  return (
+                    <tr key={integration.id}>
+                      <td>
+                        <Switch variant='ontime' {...register(`subscriptions.${index}.enabled`)} />
+                      </td>
+                      <td className={style.autoWidth}>
+                        <Select
+                          size='sm'
+                          variant='ontime'
+                          className={style.fitContents}
+                          {...register(`subscriptions.${index}.cycle`)}
+                        >
+                          {cycles.map((cycle) => (
+                            <option key={cycle.id} value={cycle.value}>
+                              {cycle.label}
+                            </option>
+                          ))}
+                        </Select>
+                      </td>
+                      <td className={style.fullWidth}>
+                        <Input
+                          size='sm'
+                          variant='ontime-filled'
+                          autoComplete='off'
+                          placeholder='http://third-party/vt1/{{timer.current}}'
+                          {...register(`subscriptions.${index}.message`, {
+                            required: { value: true, message: 'Required field' },
+                            pattern: {
+                              value: startsWithHttp,
+                              message: 'HTTP messages should start with http://',
+                            },
+                          })}
+                        />
+                        {maybeError && <Panel.Error>{maybeError}</Panel.Error>}
+                      </td>
+                      <td>
+                        <IconButton
+                          size='sm'
+                          variant='ontime-ghosted'
+                          color='#FA5656' // $red-500
```

Comment:
An improvement to keep in mind for the future would be to create an `enum` or something like that to avoid hard coding colors throughout the codebase.

Reply:
we could even use a script to sync our color scss file into a TS enum so we if we change any color it'll propagate the differences into our enum file, like

```
import fs from 'fs';
import path from 'path';

const inputFilePath = path.join('apps', 'client', 'src', 'theme', '_ontimeColours.scss');
const outputFilePath = path.join('apps', 'client', 'src', 'theme', 'colorsEnum.ts');

fs.readFile(inputFilePath, 'utf8', (err, data) => {
  if (err) {
    console.error(err);
    return;
  }

  const regex = /\$(\w+[-\d]*):\s*(.*?);/g;
  let match;
  let enumEntries = [];

  while ((match = regex.exec(data)) !== null) {
    const [_, name, value] = match;
    const enumName = name.replace(/-\d+/g, (match) => match.toUpperCase().replace('-', ''));
    enumEntries.push(`${enumName} = '${value}'`);
  }

  const enumString = `export enum Colors {\n  ${enumEntries.join(',\n  ')}\n}`;

  fs.writeFile(outputFilePath, enumString, 'utf8', (err) => {
    if (err) {
      console.error(err);
      return;
    }
    console.log('Color enum file has been generated');
  });
});
```

Reply:
Thank you @biancahpp and @arymoraes  for the input, this is a good idea.

Had a quick google on the subject and it seems that there is a standard for this called [ICSS](https://github.com/css-modules/icss) (Interoperable CSS)
We might be able to configure vite to help us here, would one one of like to give this a go?
https://github.com/vitejs/vite/discussions/13993